src/compiler/parser.ts

```diff
@@ -7318,7 +7332,10 @@ namespace ts {
                 let tags: JSDocTag[];
                 let tagsPos: number;
                 let tagsEnd: number;
-                const comments: string[] = [];
+                let linkEnd: number;
+                let commentsPos: number | undefined;
+                let comments: string[] = [];

```

Comment:
Not new, but why is `comments` accumulated as an array of parts and joined later, rather than simply accumulated as a string?

Reply:
Habit from immutable-string languages without optimisations for `+=`. I'm *not* sure if JS is still one of these languages or not.

Reply:
I'm pretty sure I remember reading that `+=` is the fastest way to accumulate a string in JS, but I don't have a source.

Reply:
I'm going to try this in a separate PR.